src/interface_adapter/settings/SettingsPresenter.java

```diff
@@ -0,0 +1,47 @@
+package interface_adapter.settings;
+
+import interface_adapter.ViewManagerModel;
+import interface_adapter.logged_in.LoggedInViewModel;
+import use_case.settings.SettingsOutputBoundary;
+import use_case.settings.SettingsOutputData;
+import use_case.switch_view.SwitchViewOutputBoundary;
+import use_case.switch_view.SwitchViewOutputData;
+
+public class SettingsPresenter implements SettingsOutputBoundary, SwitchViewOutputBoundary {
+
+  private final SettingsViewModel settingsViewModel;
+  private final LoggedInViewModel loggedInViewModel;
+  private final ViewManagerModel viewManagerModel;
+
+  public SettingsPresenter(
+      ViewManagerModel viewManagerModel,
+      LoggedInViewModel loggedInViewModel,
+      SettingsViewModel settingsViewModel) {
+    this.viewManagerModel = viewManagerModel;
+    this.loggedInViewModel = loggedInViewModel;
+    this.settingsViewModel = settingsViewModel;
+  }
+
+  @Override
+  public void prepareSuccessView(SettingsOutputData response) {
+    // On success, switch back to the same view for now
+    System.out.println("success");
+    this.viewManagerModel.setActiveView(settingsViewModel.getViewName());
```

Comment:
You shouldn't need to switch to the settings view when it succeeds right because you should already be on the settings view and just clicking the save button?